lib/rules/max-params.js

```diff
@@ -53,6 +53,11 @@ module.exports = {
 								description:
 									"Whether to count a `this` declaration when the type is `void`.",
 							},
+							countThis: {
+								enum: ["never", "except-void", "always"],
+								description:
+									"Whether to count a `this` declaration.",
+							},
 						},
 						additionalProperties: false,
```

Comment:
```suggestion
						additionalProperties: false,
						not: { required: ["countVoidThis", "countThis"] },
```

We can add this to make these options mutually exclusive.